internal/controller/controlplane_controller.go

```diff
@@ -161,7 +162,11 @@ func (r *ControlPlaneReconciler) Reconcile(ctx context.Context, req ctrl.Request
 	case tenancyv1alpha1.ControlPlaneTypeHost:
 		reconciler = host.New(r.Client, r.Scheme, r.Version, r.ClientSet, r.DynamicClient, r.EventRecorder)
 	case tenancyv1alpha1.ControlPlaneTypeExternal:
-		reconciler = external.New(r.Client, r.Scheme, r.Version, r.ClientSet, r.DynamicClient, r.EventRecorder)
+		reconciler := external.New(r.Client, r.Scheme, r.Version, r.ClientSet, r.DynamicClient, r.EventRecorder)
+		return reconciler.Reconcile(ctx, hcp)
```

Comment:
There is some change in the way the reconciler for external cluster is now created and invoked - could this be the cause of the failing e2e test for external CP ?